### ISSUE TITLE
HEAD requests every 4096 bytes

### ISSUE THREAD

**Issue by cbeckr** [s3fs-fuse/s3fs-fuse#370]
Doing a simple cp operation using current master (d6e6eebb95697f4fe184a30012848189ff9900b3):
```cp 4.mp4 /mnt/s3/4.mp4```

yields a ```HEAD``` request with every block written by ```cp``` (every 4096 bytes).

fstab:
s3fs#bucket /mnt/s3 fuse _netdev,noauto,allow_other,default_acl=public-read,dbglevel=dbg,curldbg,retries=100,parallel_count=2,enable_noobj_cache,use_cache=/tmp/s3fs-cache,stat_cache_expire=60 0 0

Log:

```
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(1829): [path=/4.mp4][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(726): [path=/4.mp4][fd=7][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Dup(711): [path=/4.mp4][fd=7][refcnt=2]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Write(1494): [path=/4.mp4][fd=7][offset=12644352][size=4096]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Load(1021): [path=/4.mp4][fd=7][offset=0][size=12644352]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(1921): [ent->file=/4.mp4][ent->fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(687): [path=/4.mp4][fd=7][refcnt=1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: [path=/4.mp4][name=security.capability][value=(nil)][size=0]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: s3fs.cpp:check_parent_object_access(656): [path=/4.mp4]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: s3fs.cpp:check_object_access(550): [path=/]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: s3fs.cpp:get_object_attribute(404): [path=/]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: s3fs.cpp:get_object_attribute(404): [path=/4.mp4]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]:       [tpath=/4.mp4]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]:       [tpath=/4.mp4][bpath=][save=][sseckeypos=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]:       URL is http://s3-eu-west-1.amazonaws.com/bucket/4.mp4
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]:       URL changed is http://bucket.s3-eu-west-1.amazonaws.com/4.mp4
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]:       computing signature [HEAD] [/4.mp4] [] []
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]:       url is http://s3-eu-west-1.amazonaws.com
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: curl.cpp:RequestPerform(1736): connecting to URL http://bucket.s3-eu-west-1.amazonaws.com/4.mp4
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: * About to connect() to bucket.s3-eu-west-1.amazonaws.com port 80 (#0)
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: *   Trying 54.231.131.56... 
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: * connected
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: > HEAD /4.mp4 HTTP/1.1#015#012Accept: */*#015#012Authorization: AWS4-HMAC-SHA256 Credential=AKXXXXXXXXXXXXXXXXXX/20160303/eu-west-1/s3/aws4_request, SignedHeaders=host;x-amz-c
ontent-sha256;x-amz-date, Signature=ef06ba3c05f9cd226e4e3c745ef729fafb80cac8821c7cdc1a0937274ec039c4#015#012host: bucket.s3-eu-west-1.amazonaws.com#015#012x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae
41e4649b934ca495991b7852b855#015#012x-amz-date: 20160303T160351Z#015#012#015
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: < HTTP/1.1 200 OK#015
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: < x-amz-id-2: ZOg6xSbigPtaKGYalh/83Z/Qs2/E8yIK1Gom6iANuWn0KHG01752bTFbm9kyI8ZkbgdT9VQnHgE=#015
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: < x-amz-request-id: C1A7A6B9C696BC29#015
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: < Date: Thu, 03 Mar 2016 16:03:52 GMT#015
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: < x-amz-meta-gid: 0#015
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: < x-amz-meta-mtime: 1457021019#015
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: < x-amz-meta-uid: 0#015
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: < x-amz-meta-mode: 33152#015
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: < Last-Modified: Thu, 03 Mar 2016 16:03:40 GMT#015
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: < ETag: "d41d8cd98f00b204e9800998ecf8427e"#015
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: < Accept-Ranges: bytes#015
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: < Content-Type: video/mp4#015
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: < Content-Length: 0#015
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: < Server: AmazonS3#015
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: < #015
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: * Connection #0 to host bucket.s3-eu-west-1.amazonaws.com left intact
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]:       HTTP response code 200
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: * Closing connection #0
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]:       add stat cache entry[path=/4.mp4]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]:       delete stat cache entry[path=/4.mp4]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: s3fs.cpp:s3fs_write(2096): [path=/4.mp4][size=4096][offset=12648448][fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:ExistOpen(1880): [path=/4.mp4][fd=7][ignore_existfd=false]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(1829): [path=/4.mp4][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(726): [path=/4.mp4][fd=7][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Dup(711): [path=/4.mp4][fd=7][refcnt=2]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Write(1494): [path=/4.mp4][fd=7][offset=12648448][size=4096]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Load(1021): [path=/4.mp4][fd=7][offset=0][size=12648448]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(1921): [ent->file=/4.mp4][ent->fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(687): [path=/4.mp4][fd=7][refcnt=1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: s3fs.cpp:s3fs_write(2096): [path=/4.mp4][size=4096][offset=12652544][fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:ExistOpen(1880): [path=/4.mp4][fd=7][ignore_existfd=false]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(1829): [path=/4.mp4][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(726): [path=/4.mp4][fd=7][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Dup(711): [path=/4.mp4][fd=7][refcnt=2]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Write(1494): [path=/4.mp4][fd=7][offset=12652544][size=4096]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Load(1021): [path=/4.mp4][fd=7][offset=0][size=12652544]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(1921): [ent->file=/4.mp4][ent->fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(687): [path=/4.mp4][fd=7][refcnt=1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: s3fs.cpp:s3fs_write(2096): [path=/4.mp4][size=4096][offset=12656640][fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:ExistOpen(1880): [path=/4.mp4][fd=7][ignore_existfd=false]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(1829): [path=/4.mp4][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(726): [path=/4.mp4][fd=7][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Dup(711): [path=/4.mp4][fd=7][refcnt=2]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Write(1494): [path=/4.mp4][fd=7][offset=12656640][size=4096]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Load(1021): [path=/4.mp4][fd=7][offset=0][size=12656640]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(1921): [ent->file=/4.mp4][ent->fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(687): [path=/4.mp4][fd=7][refcnt=1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: s3fs.cpp:s3fs_write(2096): [path=/4.mp4][size=4096][offset=12660736][fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:ExistOpen(1880): [path=/4.mp4][fd=7][ignore_existfd=false]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(1829): [path=/4.mp4][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(726): [path=/4.mp4][fd=7][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Dup(711): [path=/4.mp4][fd=7][refcnt=2]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Write(1494): [path=/4.mp4][fd=7][offset=12660736][size=4096]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Load(1021): [path=/4.mp4][fd=7][offset=0][size=12660736]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(1921): [ent->file=/4.mp4][ent->fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(687): [path=/4.mp4][fd=7][refcnt=1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: s3fs.cpp:s3fs_write(2096): [path=/4.mp4][size=4096][offset=12664832][fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:ExistOpen(1880): [path=/4.mp4][fd=7][ignore_existfd=false]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(1829): [path=/4.mp4][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(726): [path=/4.mp4][fd=7][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Dup(711): [path=/4.mp4][fd=7][refcnt=2]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Write(1494): [path=/4.mp4][fd=7][offset=12664832][size=4096]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Load(1021): [path=/4.mp4][fd=7][offset=0][size=12664832]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(1921): [ent->file=/4.mp4][ent->fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(687): [path=/4.mp4][fd=7][refcnt=1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: s3fs.cpp:s3fs_write(2096): [path=/4.mp4][size=4096][offset=12668928][fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:ExistOpen(1880): [path=/4.mp4][fd=7][ignore_existfd=false]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(1829): [path=/4.mp4][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(726): [path=/4.mp4][fd=7][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Dup(711): [path=/4.mp4][fd=7][refcnt=2]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Write(1494): [path=/4.mp4][fd=7][offset=12668928][size=4096]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Load(1021): [path=/4.mp4][fd=7][offset=0][size=12668928]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(1921): [ent->file=/4.mp4][ent->fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(687): [path=/4.mp4][fd=7][refcnt=1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: s3fs.cpp:s3fs_write(2096): [path=/4.mp4][size=4096][offset=12673024][fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:ExistOpen(1880): [path=/4.mp4][fd=7][ignore_existfd=false]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(1829): [path=/4.mp4][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(726): [path=/4.mp4][fd=7][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Dup(711): [path=/4.mp4][fd=7][refcnt=2]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Write(1494): [path=/4.mp4][fd=7][offset=12673024][size=4096]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Load(1021): [path=/4.mp4][fd=7][offset=0][size=12673024]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(1921): [ent->file=/4.mp4][ent->fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(687): [path=/4.mp4][fd=7][refcnt=1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: s3fs.cpp:s3fs_write(2096): [path=/4.mp4][size=4096][offset=12677120][fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:ExistOpen(1880): [path=/4.mp4][fd=7][ignore_existfd=false]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(1829): [path=/4.mp4][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Open(726): [path=/4.mp4][fd=7][size=-1][time=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Dup(711): [path=/4.mp4][fd=7][refcnt=2]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Write(1494): [path=/4.mp4][fd=7][offset=12677120][size=4096]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Load(1021): [path=/4.mp4][fd=7][offset=0][size=12677120]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(1921): [ent->file=/4.mp4][ent->fd=7]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: fdcache.cpp:Close(687): [path=/4.mp4][fd=7][refcnt=1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: [path=/4.mp4][name=security.capability][value=(nil)][size=0]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: s3fs.cpp:check_parent_object_access(656): [path=/4.mp4]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: s3fs.cpp:check_object_access(550): [path=/]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: s3fs.cpp:get_object_attribute(404): [path=/]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: s3fs.cpp:get_object_attribute(404): [path=/4.mp4]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]:       [tpath=/4.mp4]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]:       [tpath=/4.mp4][bpath=][save=][sseckeypos=-1]
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]:       URL is http://s3-eu-west-1.amazonaws.com/bucket/4.mp4
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]:       URL changed is http://bucket.s3-eu-west-1.amazonaws.com/4.mp4
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]:       computing signature [HEAD] [/4.mp4] [] []
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]:       url is http://s3-eu-west-1.amazonaws.com
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: curl.cpp:RequestPerform(1736): connecting to URL http://bucket.s3-eu-west-1.amazonaws.com/4.mp4
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: * About to connect() to bucket.s3-eu-west-1.amazonaws.com port 80 (#0)
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: *   Trying 54.231.131.56... 
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: * connected
Mar  3 17:03:51 ip-10-0-1-14 s3fs[7391]: > HEAD /4.mp4 HTTP/1.1#015#012Accept: */*#015#012Authorization: AWS4-HMAC-SHA256 Credential=AKXXXXXXXXXXXXXXXXXX/20160303/eu-west-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=ef06ba3c05f9cd226e4e3c745ef729fafb80cac8821c7cdc1a0937274ec039c4#015#012host: bucket.s3-eu-west-1.amazonaws.com#015#012x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855#015#012x-amz-date: 20160303T160351Z#015#012#015
```

**Comment by RobbKistler** [comment]
I see 2 HEAD requests and ~9 4096 byte calls into s3fs_write.  I don't see a PUT of the object, either.  Can you add some detail on why you think the HEAD is happening every 4k?

**Comment by cbeckr** [comment]
Sorry, I should have mentioned that this is just a brief excerpt - the HEAD happens about twice per second, hence the full log spans several GB.
The PUT actually never happens for large files if the cache is enabled (the file will be 0 bytes on S3). But this is probably a separate issue.

**Comment by RobbKistler** [comment]
I see this interesting log before the HEAD: 

```
 s3fs[7391]: [path=/4.mp4][name=security.capability][value=(nil)][size=0]
```

That log line is generated by s3fs_getxattr(), which gets called when the file is accessed with getfattr. Something is looking for the extended attributed named "security.capability".  I don't see anywhere this would be called internally by s3fs-fuse.

Is it possible something other than the cp process is accessing the file?  I haven't seen this before, so I'm just guessing. 

If you start s3fs-fuse from the command line instead of fstab, you can add "-d -d".  This will enable debug messages from FUSE.  Look for a line like the following before the HEAD.  It would interesting to know if the pid is that of the cp process, or something else. 

```unique: 2, opcode: GETXATTR (3), nodeid: 1, insize: 56, pid: 23336```

**Comment by ggtakec** [comment]
Hi, @cbeckr 
I had reproduced this.
By that supports xattr, it seems to confirm the security.capability every 64KB from FUSE(OS?).

If s3fs uses the stat_cache, s3fs does not send the request. But s3fs_getxattr callback function is called every 64KB.

I will check the codes and logic about implementation of xattr support, and I think this logic should be optinal.

Please wait a while and if you can please set max_stat_cache_size option.

@RobbKistler 
Thanks for your help.

Regards,

**Comment by ggtakec** [comment]
Hi @cbeckr 
I merged #371 for this issue.(you can see detail #371 comments)
If you can, please try to use latest master branch codes.

Regards,
